core/Azure.Mcp.Core/tests/Azure.Mcp.Tests/Client/RecordedCommandTestsBase.cs

```diff
@@ -107,6 +126,77 @@ public override async ValueTask InitializeAsync()
 
         // start recording/playback session
         await StartRecordOrPlayback();
+
+        // apply custom matcher if test has attribute
+        await ApplyAttributeMatcherSettings();
+    }
+
+    private async Task ApplyAttributeMatcherSettings()
+    {
+        if (Proxy == null || TestMode != TestMode.Playback)
+        {
+            return;
+        }
+
+        var methodName = TestContext.Current?.TestMethod?.MethodName;
+        if (string.IsNullOrEmpty(methodName))
+        {
+            return;
+        }
+
+        // Use reflection to find the test method on the derived class
+        var testMethod = GetType().GetMethod(methodName);
+        if (testMethod == null)
+        {
+            return;
+        }
```

Comment:
This seems fairly indirect.  The docs show TestContext.TestMethod having a `Method` property with type `MethodInfo`. is that not the case?
https://api.xunit.net/v3/3.2.0/Xunit.v3.IXunitTestMethod.html

Reply:
It's not populated properly in my experience, no. I'll try again?